### PR TITLE
Try combining some steps to speed up CI

### PR DESCRIPTION
**Pull request by therealprof**
Signed-off-by: Daniel Egger <daniel@eggers-club.de>

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -27,10 +24,20 @@ jobs:
         with:
           profile: minimal
           toolchain: ${{ matrix.rust }}
-          target: ${{ matrix.TARGET }}
+          target: thumbv6m-none-eabi thumbv7m-none-eabi thumbv7em-none-eabi thumbv7em-none-eabihf thumbv8m.main-none-eabi
```
Comment (jonas-schievink) [inline .github/workflows/ci.yml:27]: I don't think this action supports lists https://github.com/actions-rs/toolchain/pull/62

Comment (jonas-schievink) [reply .github/workflows/ci.yml:27]: It might be easiest to just invoke rustup manually, it *can* take a list of targets to install

Comment (therealprof) [reply .github/workflows/ci.yml:27]: Yes indeed. I'm alreay working on it.

**Comment by therealprof** [comment]
Hm, no benefit here wall clock wise. But we're using way less builders which might be benefitial anyway.

**Comment by jonas-schievink** [comment]
We could also collapse the Clippy and rustfmt builders into one

**Comment by therealprof** [comment]
Yes we could but I actually prefer them to be separate because that makes it easier for me to add them to other crates. 😅

**Pull request merged**